plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java

```diff
@@ -313,66 +388,138 @@ public void handle(final HttpExchange exchange) throws IOException {
                     exchange.sendResponseHeaders(RestStatus.OK.getStatus(), response.length);
                     exchange.getResponseBody().write(response);
 
-                } else if (Regex.simpleMatch("POST /upload/storage/v1/b/bucket/*uploadType=multipart*", request)) {
-                    byte[] response = new byte[0];
-                    try (BufferedInputStream in = new BufferedInputStream(new GZIPInputStream(exchange.getRequestBody()))) {
-                        String blob = null;
-                        int read;
-                        while ((read = in.read()) != -1) {
-                            boolean markAndContinue = false;
-                            try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-                                do { // search next consecutive {carriage return, new line} chars and stop
-                                    if ((char) read == '\r') {
-                                        int next = in.read();
-                                        if (next != -1) {
-                                            if (next == '\n') {
-                                                break;
+                } else if (Regex.simpleMatch("POST /upload/storage/v1/b/bucket/*uploadType=*", request)) {
+                    final Map<String, String> params = new HashMap<>();
+                    RestUtils.decodeQueryString(exchange.getRequestURI().getQuery(), 0, params);
+                    final String uploadType = params.get("uploadType");
+
+                    if (uploadType.equals("resumable")) {
```

Comment:
I could maybe avoid this extra condition and make the code easier to read by using two different regex:

```
} else if (Regex.simpleMatch("POST /upload/storage/v1/b/bucket/*uploadType=resumable*", request)) {
...
} else if (Regex.simpleMatch("POST /upload/storage/v1/b/bucket/*uploadType=multipart*", request)) {
...
```